**Soal 1. Data Preparation**

Bacalah Dataset dengan memberikan beberapa argumen sehingga bisa membaca datetime

In [28]:
#code here
import pandas as pd
df = pd.read_csv("climate_change.csv", parse_dates=True, index_col="date")
df

meantemp    humidity  wind_speed  meanpressure
date                                                       
2013-01-01  10.000000   84.500000    0.000000   1015.666667
2013-01-02   7.400000   92.000000    2.980000   1017.800000
2013-01-03   7.166667   87.000000    4.633333   1018.666667
2013-01-04   8.666667   71.333333    1.233333   1017.166667
2013-01-05   6.000000   86.833333    3.700000   1016.500000
...               ...         ...         ...           ...
2016-12-28  17.217391   68.043478    3.547826   1015.565217
2016-12-29  15.238095   87.857143    6.000000   1016.904762
2016-12-30  14.095238   89.666667    6.266667   1017.904762
2016-12-31  15.052632   87.000000    7.325000   1016.100000
2017-01-01  10.000000  100.000000    0.000000   1016.000000

[1462 rows x 4 columns]

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1TqrXs9JStRXqiF0NIWiMdo5r8CQ-S74-)



---



---



Soal 2. Selection

Tampilkan data keseluruhan untuk tahun 2015 di bulan November

In [29]:
#code here
df["2015-11-1":"2015-11-30"]

meantemp  humidity  wind_speed  meanpressure
date                                                    
2015-11-01    22.875    70.500      0.9250      1016.250
2015-11-02    23.375    71.375      0.4625      1015.875
2015-11-03    24.000    67.875      0.4750      1014.250
2015-11-04    23.750    69.000      0.9250      1012.250
2015-11-05    22.125    75.250      5.1000      1012.375
2015-11-06    22.875    74.375      2.3250      1012.750
2015-11-07    21.875    74.875      1.1625      1014.625
2015-11-08    21.875    70.250      1.8625      1015.875
2015-11-09    21.750    73.625      0.7000      1015.000
2015-11-10    22.750    72.625      0.7000      1013.500
2015-11-11    21.875    59.500      4.6375      1014.375
2015-11-12    21.875    60.500      5.7875      1015.375
2015-11-13    22.375    59.375      5.3250      1014.875
2015-11-14    22.375    59.625      3.4750      1012.000
2015-11-15    22.500    62.000      0.0000      1012.000
2015-11-16    21.875    50.375      2.7875      1012.625
2015-11-17    19.875    48.375      5.1000      1013.500
2015-11-18    18.500    57.625      4.4000      1016.375
2015-11-19    18.500    64.750      0.9250      1017.500
2015-11-20    18.500    66.000      0.4625      1016.125
2015-11-21    18.750    67.625      1.3875      1015.500
2015-11-22    18.625    64.875      0.4625      1015.125
2015-11-23    17.000    73.500      0.0000      1015.875
2015-11-24    18.500    63.000      1.6250      1016.000
2015-11-25    19.375    56.375      5.7875      1014.750
2015-11-26    20.000    59.500      0.9250      1012.750
2015-11-27    18.500    66.250      2.5500      1011.250
2015-11-28    18.750    82.125      1.1625      1015.375
2015-11-29    18.125    81.250      1.1625      1018.000
2015-11-30    19.500    82.375      0.9250      1016.750

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1ABMLSmdQ817PIdxW1v7rUKelOj_W241g)



---



---



**Soal 3. Filtering**

Tampilkan data perubahan iklim pada 1 bulan terakhir di tahun 2016 dan jawablah pertanyaan berikut:

Berapakah rata-rata temperatur tertinggi?

In [48]:
#code here
df_december = df["2016-01-01":"2016-12-31"].last("1M")

C:\Users\Krisna Sandy Pribadi\AppData\Local\Temp\ipykernel_1052\2313461350.py:2: FutureWarning: last is deprecated and will be removed in a future version. Please create a mask and filter using `.loc` instead
  df_december = df["2016-01-01":"2016-12-31"].last("1M")


Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1X2hXsvrRGvqHqtc5BFbVvRXwh5YxBQ_P)

In [49]:
df_december.mean()

meantemp          17.672565
humidity          68.840384
wind_speed         6.132705
meanpressure    1016.290993
dtype: float64

Jawab disini: Rata-rata temperatur 17.672565



---



---



**Soal 4. Time grouping**

 

*   Lakukan grouping data perubahan iklim pada tahun 2016 berdasarkan frekuensi 1 bulan dengan memberikan agregasi max pada kolom meantemp. humidity, wind_speed dan meanpressure!
*   Berikan analisa, di bulan apa rata-rata temperatur, humidity, wind_speed dan meanpressure tertinggi?





In [53]:
#code here
group_2016_max = df["2016-01-01":"2016-12-31"].groupby(pd.Grouper(freq="M")).agg("max")
group_2016_max

meantemp   humidity  wind_speed  meanpressure
date                                                      
2016-01-31  20.142857  88.833333   11.240000   1021.142857
2016-02-29  23.875000  77.000000   17.137500   1020.071429
2016-03-31  30.000000  67.800000   22.100000   7679.333333
2016-04-30  35.687500  44.250000   19.653846   1009.400000
2016-05-31  38.272727  74.812500   18.425000   1007.466667
2016-06-30  38.200000  75.071429   21.306250   1002.833333
2016-07-31  34.187500  93.875000   13.656250   1004.625000
2016-08-31  33.809524  89.555556   14.666667   1006.200000
2016-09-30  33.444444  87.240000   17.908333   1352.615385
2016-10-31  33.269231  71.875000   15.825000   1017.807692
2016-11-30  25.640000  80.086957   15.085000   1350.296296
2016-12-31  21.208333  94.300000   19.404762   1019.700000

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1s2Y8Dc3M3xbG-51mTuTmXUFgLvHDvc5Q)

In [70]:
max_meantemp = group_2016_max["meantemp"] == group_2016_max["meantemp"].max()
max_humidity = group_2016_max["humidity"] == group_2016_max["humidity"].max()
max_wind_speed = group_2016_max["wind_speed"] == group_2016_max["wind_speed"].max()
max_meanpressure = group_2016_max["meanpressure"] == group_2016_max["meanpressure"].max()
display(group_2016_max[max_meantemp])
display(group_2016_max[max_humidity])
display(group_2016_max[max_wind_speed])
display(group_2016_max[max_meanpressure])

meantemp  humidity  wind_speed  meanpressure
date                                                     
2016-05-31  38.272727   74.8125      18.425   1007.466667

meantemp  humidity  wind_speed  meanpressure
date                                                     
2016-12-31  21.208333      94.3   19.404762        1019.7

meantemp  humidity  wind_speed  meanpressure
date                                                    
2016-03-31      30.0      67.8        22.1   7679.333333

meantemp  humidity  wind_speed  meanpressure
date                                                    
2016-03-31      30.0      67.8        22.1   7679.333333

meantemp          38.272727
humidity          94.300000
wind_speed        22.100000
meanpressure    7679.333333
dtype: float64

Jawab disini:Rata-rata meantemp tertinggi di bulan Mei, humidity tertinggi di bulan Desember, wind speed dan meanpressure tertinggi di bulan Maret



---



---



**Soal 5. Time Differenced**

Tunjukkanlah perubahan iklim di tahun 2016 pada minggu terakhir

In [72]:
#code here
df["2016-01-01":"2016-12-31"].last("1W").diff()

C:\Users\Krisna Sandy Pribadi\AppData\Local\Temp\ipykernel_1052\1457778501.py:2: FutureWarning: last is deprecated and will be removed in a future version. Please create a mask and filter using `.loc` instead
  df["2016-01-01":"2016-12-31"].last("1W").diff()


meantemp   humidity  wind_speed  meanpressure
date                                                     
2016-12-26       NaN        NaN         NaN           NaN
2016-12-27 -0.292857  -7.307143   -0.449211      0.247619
2016-12-28  0.367391   0.493478   -4.787174     -1.634783
2016-12-29 -1.979296  19.813665    2.452174      1.339545
2016-12-30 -1.142857   1.809524    0.266667      1.000000
2016-12-31  0.957393  -2.666667    1.058333     -1.804762

Expected output:

(Gambar hanya sebagian)

![alt text](https://drive.google.com/uc?id=1miJBQeGMyUS_G3bwP14Plo4s5950oYy5)



---



---

